test/ng/compileSpec.js

```diff
@@ -4827,6 +4827,31 @@ describe('$compile', function() {
             });
         });
 
+
+        it('should reassign bindings to the controller before $onInit, in case the constructor has overwritten them', function() {
+          angular.module('owComponentTest', [])
+            .component('owComponent', {
+              bindings: { input: '<' },
+              controller: function() {
+                component = this;
+                this.input = 'constructor-changed';
+                this.$onInit = function() {
+                  log.push('$onInit: input=' + this.input);
+                };
+              }
+            });
+
+          module('owComponentTest');
+          inject(function() {
+            $rootScope.name = 'outer';
+            compile('<ow-component input="name"></ow-component>');
+            expect(component.input).toEqual('outer');
```

Comment:
why? the expectation is that when onInit runs we initialize the bindings overwriting anything that the developer set in the constructor.